src/Products/Annotation/Controllers/AnnotationApiController.cs

```diff
@@ -73,357 +55,214 @@ public AnnotationConfiguration LoadConfig()
-                {
-                    documentPath = fileName;
-                }
-                else
-                {
-                    documentPath = Path.Combine(parentDirName, fileName);
-                }
-                DocumentInfoContainer documentInfo = AnnotationImageHandler.GetDocumentInfo(documentPath, password);
-                // check if document type is image
-                if (SupportedImageFormats.Contains(Path.GetExtension(documentGuid)))
+                using (GroupDocs.Annotation.Annotator annotator = new GroupDocs.Annotation.Annotator(outputStream, GetLoadOptions(password)))
                 {
-                    documentType = "image";
-                }
-                // initiate annotator object  
-                string notSupportedMessage = "";
-                for (int i = 0; i < annotationsData.Length; i++)
-                {
-                    // create annotator
-                    AnnotationDataEntity annotationData = annotationsData[i];
-                    PageData pageData = documentInfo.Pages[annotationData.pageNumber - 1];
-                    // add annotation, if current annotation type isn't supported by the current document type it will be ignored
-                    try
-                    {
-                        BaseAnnotator annotator = AnnotatorFactory.createAnnotator(annotationData, pageData);
-                        if (annotator.IsSupported(documentType))
-                        {
-                            annotations.Add(annotator.GetAnnotationInfo(documentType));
-                        }
-                        else
-                        {
-                            notSupportedMessage = annotator.Message;
-                        }
-                    }
-                    catch (System.Exception ex)
-                    {
-                        throw new System.Exception(ex.Message, ex);
-                    }
-                }
+                    PreviewOptions previewOptions = new PreviewOptions(pageNumber => result);
 
-                // Add annotation to the document
-                DocumentType type = DocumentTypesConverter.GetDocumentType(documentType);
-                RemoveAnnotations(documentGuid);
-                // check if annotations array contains at least one annotation to add
-                if (annotations.Count != 0)
-                {
-                    Stream cleanDoc = new FileStream(documentGuid, FileMode.Open, FileAccess.ReadWrite, FileShare.ReadWrite);
-                    Stream result = AnnotationImageHandler.ExportAnnotationsToDocument(cleanDoc, annotations, type);
-                    cleanDoc.Dispose();
-                    cleanDoc.Close();
-                    result.Position = 0;
-                    return result;
+                    previewOptions.PreviewFormat = PreviewFormats.PNG;
+                    previewOptions.PageNumbers = new int[] { pageNumberToRender };
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the array type; it is redundant.](https://app.codacy.com/manual/GroupDocs/GroupDocs.Annotation-for-.NET-MVC/pullRequest?prid=5372470)